### DISCUSSION TITLE
Custom Validation (Database Records)

### DISCUSSION THREAD

**Discussion by itokome** [Budibase/budibase#10521] [Help]
Can I use custom validation to check that the entered value is correct if none exists compared to a list of existing database records?
In the following cases, the value entered matches the first line, so an input error is assumed.

Input: aaabbbccc

Database:
1:aaabbbccc
2:aaaaaaaaa
3:bbbbbbbb

**Comment by applicationuser** [comment]
@itokome 

you can do that using validate form action, Please check this  https://github.com/Budibase/budibase/discussions/5143#discussioncomment-2459918

**Comment by itokome** [comment]
@applicationuser 

Thanks for the reply.
You've been very helpful.

The method you gave me is exactly what I was looking for and seems to work correctly.
But still not getting the desired results.

I am creating a screen to insert a record into Data Provider A
I want to have a constraint on the insert that it cannot insert a record that matches a record that exists in Data Provider B's master database.

I have been taught to retrieve Rows from Data Provider B, compare them with Input items on the screen, set MatchOnName if there is a match, and set in validation that Input items and MatchOnName must not be equal, but I have not been able to get the result that I want. MatchOnName is set if there is a match.
However, the data matched with the Row of Data Provider B can be registered to Data Provider A.

Is it possible to mix two Data Providers in one Form?
How can I make sure that the Rows retrieved from Data Provider B are really what we expect?
Any ideas?

**Reply by applicationuser** [reply]
@itokome 
can you share app screenshots, how you setup data provider A and data provider b.  what filters applied on each data provider. 
you can also output to paragraph row count of Data provider based on the filter to check for any issue.
